### PR TITLE
{devel}[foss/2016a] Yade 1.20.0 (REVIEW)

### PR DESCRIPTION
**Pull request by jrbosch**
support for Yade in easybuild with foss 2016a toolchian.
Needs PR ~~#2909~~ and ~~#2923~~ 

edit: requires ~~#2969, #2596~~

### CODE REVIEW

**Comment by jrbosch** [comment]
@boegel  and @hajgato this is what I want to do. but to finish it I think I will need your help in some suggestion.

**Comment by boegel** [comment]
Jenkins: ok to test

File: easybuild/easyconfigs/y/Yade/Yade-1.20.0-foss-2016a.eb [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+pyver = '2.7.11'
+versionsuffix = '-Python-%s' % pyver
+
+dependencies = [
+    ('git', '2.8.0'),
+    ('freeglut', '3.0.0'),
+    ('loki', '-------'),
+    ('Boost', '1.60.0'),
+    ('fakeroot', '-------'),
+    ('dpkg', '-------'),
```
Comment (boegel) [inline easybuild/easyconfigs/y/Yade/Yade-1.20.0-foss-2016a.eb]: This is probably not required, since this is a tool for managing packages on Debian?

**Comment by boegel** [comment]
@jrbosch Let us know what questions you have on this, it would be nice to get this included by the time EasyBuild v2.8.0 is released (mid-end May).

**Comment by boegel** [comment]
@ChristopherBignamini I was informed by @gppezzi that you have been working on an easyconfig for Yade as well?

@jrbosch If you're looking for inspiration/feedback, see https://github.com/eth-cscs/tools/blob/master/easybuild/easyconfigs/y/YADE/YADE-1.09.0-foss-2015b.eb

**Comment by jrbosch** [comment]
@boegel minieigen easyconfig is not ok, it has problems in build step it says missing lboost_python I'm not yet sure what is missing because libboost_python is present but I'll try to fix it

**Comment by jrbosch** [comment]
@boegel this PR needs PR #2909 and #2923 I think Yade is almost ready (on my server is already running), just missing me to do some tests with configopts and some dependencies, then I will change the tag to REVIEW.
Regards.

File: easybuild/easyconfigs/l/Loki/Loki-0.1.7_change_prefix.patch [hunk 2]
```diff
@@ -0,0 +1,9 @@
+diff -ru loki-0.1.7.orig/Makefile.common loki-0.1.7/Makefile.common
+--- loki-0.1.7.orig/Makefile.common	2007-05-28 22:26:26.000000000 -0400
++++ loki-0.1.7/Makefile.common	2016-04-21 18:20:54.280476472 -0400
+@@ -4,4 +4,4 @@
+ CXXWARNFLAGS := -Wall -Wold-style-cast -Wundef -Wsign-compare -Wconversion -Wpointer-arith -pedantic
+ CXXFLAGS := $(CXXWARNFLAGS) -g -O2
+ 
+-prefix := /usr
++prefix := $(EB_INSTALL_DIR)
```
Comment (boegel) [inline easybuild/easyconfigs/l/Loki/Loki-0.1.7_change_prefix.patch]: I don't think this patch is needed? Have you tried using this?

```python
installopts = "prefix=%(installdir)s"
```

or

```python
preinstallopts = "export prefix=%(installdir)s && "
```

I think that should override the `/usr`, since they're using `:=` rather than `=`?

File: easybuild/easyconfigs/m/minieigen/minieigen-0.5.3_Eigen.patch [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+-import sys, glob
++import os, sys, glob
+ 
+ # vectorization is currently broken and experimental ONLY
+ vectorize=False
+@@ -19,10 +19,11 @@
+ 	# * https://bugs.launchpad.net/panda3d/+bug/919237
+ 		define_macros+=[('EIGEN_DONT_VECTORIZE',None)]
+ else:
++	eigen_prefix = os.environ["EBROOTEIGEN"]
```
Comment (boegel) [inline easybuild/easyconfigs/m/minieigen/minieigen-0.5.3_Eigen.patch]: maybe use this instead, in case this patch is used outside of EasyBuild?

```python
```

File: easybuild/easyconfigs/y/Yade/Yade-1.20.0-foss-2016a.eb [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+ focused on Discrete Element Method. The computation parts are written in c++ using flexible object model, 
+ allowing independent implementation of new alogrithms and interfaces. 
+ Python is used for rapid and concise scene construction, simulation control, postprocessing and debugging. """
+
+toolchain = {'name': 'foss', 'version': '2016a'}
+
+source_urls = ['https://launchpad.net/yade/trunk/yade-1.00.0/+download/']
+sources = [SOURCELOWER_TAR_GZ]
+
+suffix = '-Python-%(pyver)s'
```
Comment (boegel) [inline easybuild/easyconfigs/y/Yade/Yade-1.20.0-foss-2016a.eb]: make this `versionsuffix`, move it up below `version =` and rename the easyconfig to `Yade-1.20.0-foss-2016a.eb-Python-2.7.11`?

File: easybuild/easyconfigs/y/Yade/Yade-1.20.0-foss-2016a.eb [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    ('Qt', '4.8.7'),
+    ('GTS', '0.7.6'),
+    ('Python', '2.7.11'),
+    ('libQGLViewer', '2.6.3'),
+#    ('GDB', '7.10.1',suffix),
+    # optional
+#    ('VTK', '6.3.0'),			
+#    ('CGAL', '4.7'),
+#    ('SuiteSparse', '4.5.1'),
+#    ('OpenBLAS', '0.2.15'),
```
Comment (boegel) [inline easybuild/easyconfigs/y/Yade/Yade-1.20.0-foss-2016a.eb]: this is already part of `foss`

File: easybuild/easyconfigs/y/Yade/Yade-1.20.0-foss-2016a.eb [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+# ENABLE_MASK_ARBITRARY: enable MASK_ARBITRARY option (OFF by default)
+# ENABLE_PROFILING: enable profiling, e.g., shows some more metrics, which can define bottlenecks of the code (OFF by default)
+# runtimePREFIX: used for packaging, when install directory is not the same as runtime directory (/usr/local by default)
+# CHUNKSIZE: specifiy the chunk size if you want several sources to be compiled at once. Increases compilation speed but RAM-consumption during compilation as well (1 by default)
+# VECTORIZE: enables vectorization and alignment in Eigen3 library, experimental (OFF by default)
+# USE_QT5: use QT5 for GUI, experimental (OFF by default)
+
+sanity_check_paths = {
+    'files': ['bin/yade-%(version)s', 'bin/yade-%(version)s-batch', 'lib/x86_64-linux-gnu/yade-%%(version)s/libyade.%s' % SHLIB_EXT],
+    'dirs': ['lib/x86_64-linux-gnu/yade-%(versio)s/py/yade'],
```
Comment (boegel) [inline easybuild/easyconfigs/y/Yade/Yade-1.20.0-foss-2016a.eb]: typo here: `%(version)s` (you're missing the `n`)

**Comment by boegel** [comment]
@jrbosch #2909 and #2923 ar merged now, let me know if you need any help with this

Try not to make this PR much bigger (in terms of number of easyconfig files included).

**Comment by jrbosch** [comment]
@boegel I think that the last missing dependency for Yade is CGAL and this exist in the the PR #2596 , so when CGAL became completed I will prepare it for foss

File: easybuild/easyconfigs/y/Yade/Yade-1.20.0-foss-2016a.eb (hunk 4 above)
Comment (boegel) [inline easybuild/easyconfigs/y/Yade/Yade-1.20.0-foss-2016a.eb]: @jrbosch you'll need to remove this easyconfig, since you added one that includes the `-Python` versionsuffix...

File: easybuild/easyconfigs/y/Yade/Yade-1.20.0-foss-2016a-Python-2.7.11.eb [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+
+toolchain = {'name': 'foss', 'version': '2016a'}
+
+source_urls = ['https://launchpad.net/yade/trunk/yade-1.00.0/+download/']
+sources = [SOURCELOWER_TAR_GZ]
+
+dependencies = [
+    ('Loki', '0.1.7'),
+    ('Boost', '1.60.0', versionsuffix),
+    ('GL2PS', '1.3.9'),
```
Comment (boegel) [inline easybuild/easyconfigs/y/Yade/Yade-1.20.0-foss-2016a-Python-2.7.11.eb]: this depends on Mesa (v11.1.2), so we'll need to make a variant of this that depends on Mesa 11.2.1, to make it compatible with the Mesa underneath CGAL 3.8 (cfr. #2596)

File: easybuild/easyconfigs/y/Yade/Yade-1.20.0-foss-2016a-Python-2.7.11.eb [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+    ('Python', '2.7.11'),    
+    ('IPython', '3.2.3', versionsuffix),
+    ('PyGTS', '0.3.1', versionsuffix),
+    ('Eigen', '3.2.7'),
+    ('minieigen', '0.5.3', versionsuffix),
+    ('Qt', '4.8.7'),
```
Comment (boegel) [inline easybuild/easyconfigs/y/Yade/Yade-1.20.0-foss-2016a-Python-2.7.11.eb]: this depends on GLib 2.47.5, we'll need to make a variant that uses GLib 2.48.0 to be compatible with *GTS and CGAL 4.8...

File: easybuild/easyconfigs/y/Yade/Yade-1.20.0-foss-2016a-Python-2.7.11.eb [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+    ('GTS', '0.7.6'),
+    ('libQGLViewer', '2.6.3'),
```
Comment (boegel) [inline easybuild/easyconfigs/y/Yade/Yade-1.20.0-foss-2016a-Python-2.7.11.eb]: this depends on libGLU 9.0.0, which depends on Mesa 11.1.2, so again, we'll need to make a variant of this that depends on the libGLU that uses Mesa 11.2.1, cfr. #2977

File: easybuild/easyconfigs/y/Yade/Yade-1.20.0-foss-2016a-Python-2.7.11.eb [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+    ('GDB', '7.10.1', versionsuffix),
+    ('zlib', '1.2.8'),
+    # optional
+#    ('CGAL', '4.7'),
```
Comment (boegel) [inline easybuild/easyconfigs/y/Yade/Yade-1.20.0-foss-2016a-Python-2.7.11.eb]: bum pto CGAL 3.8, and include the versionsuffix, cfr. #2596

File: easybuild/easyconfigs/y/Yade/Yade-1.20.0-foss-2016a-Python-2.7.11.eb [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+]
+
+builddependencies = [('CMake', '3.4.3')]
```
Comment (boegel) [inline easybuild/easyconfigs/y/Yade/Yade-1.20.0-foss-2016a-Python-2.7.11.eb]: bump to latest 3.5.2 (although it doesn't matter than much)?

File: easybuild/easyconfigs/y/Yade/Yade-1.20.0-foss-2016a-Python-2.7.11.eb [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+configopts += '-DCGAL_INCLUDE_DIR=$EBROOTCGAL/include '
+configopts += '-DGMP_INCLUDE_DIR=$EBROOTGMP/include '
+configopts += '-DGMP_LIBRARIES=$EBROOTGMP/lib '
+configopts += '-DGMPXX_LIBRARIES=$EBROOTGMP/lib '
+
+#to disable CGAL uncomment the following line:
+#configopts += '-DENABLE_CGAL=OFF '
+
+sanity_check_paths = {
+    'files': ['bin/yade-%(version)s', 'bin/yade-%(version)s-batch', 'lib/x86_64-linux-gnu/yade-%%(version)s/libyade.%s' % SHLIB_EXT],
```
Comment (boegel) [inline easybuild/easyconfigs/y/Yade/Yade-1.20.0-foss-2016a-Python-2.7.11.eb]: @jrbosch line too long, should be less than 120 characters, so please use:

```python
sanity_check_paths = {
    'files': ['bin/yade-%(version)s', 'bin/yade-%(version)s-batch',
              'lib/x86_64-linux-gnu/yade-%%(version)s/libyade.%s' % SHLIB_EXT],
```

File: easybuild/easyconfigs/y/Yade/Yade-1.20.0-foss-2016a-Python-2.7.11.eb [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+configopts = '-DENABLE_GUI=OFF '
+configopts += '-DENABLE_VTK=OFF '
+configopts += '-DENABLE_LINSOLV=OFF '
+configopts += '-DENABLE_PFVFLOW=OFF '
+configopts += '-DCGAL_INCLUDE_DIR=$EBROOTCGAL/include '
+configopts += '-DGMP_INCLUDE_DIR=$EBROOTGMP/include '
+configopts += '-DGMP_LIBRARIES=$EBROOTGMP/lib '
+configopts += '-DGMPXX_LIBRARIES=$EBROOTGMP/lib '
+
+#to disable CGAL uncomment the following line:
```
Comment (boegel) [inline easybuild/easyconfigs/y/Yade/Yade-1.20.0-foss-2016a-Python-2.7.11.eb]: @jrbosch any particular reason you include this? Why would you want to disable CGAL?

File: easybuild/easyconfigs/y/Yade/Yade-1.20.0-foss-2016a-Python-2.7.11.eb [hunk 14]
```diff
@@ -0,0 +1,55 @@
...
+separate_build_dir = True
+
+configopts = '-DENABLE_GUI=OFF '
+configopts += '-DENABLE_VTK=OFF '
+configopts += '-DENABLE_LINSOLV=OFF '
+configopts += '-DENABLE_PFVFLOW=OFF '
+configopts += '-DCGAL_INCLUDE_DIR=$EBROOTCGAL/include '
+configopts += '-DGMP_INCLUDE_DIR=$EBROOTGMP/include '
+configopts += '-DGMP_LIBRARIES=$EBROOTGMP/lib '
+configopts += '-DGMPXX_LIBRARIES=$EBROOTGMP/lib '
```
Comment (boegel) [inline easybuild/easyconfigs/y/Yade/Yade-1.20.0-foss-2016a-Python-2.7.11.eb]: maybe shorten this a bit by grouping related things?

```python
configopts = '-DENABLE_GUI=OFF -DENABLE_VTK=OFF -DENABLE_LINSOLV=OFF -DENABLE_PFVFLOW=OFF '
configopts += '-DCGAL_INCLUDE_DIR=$EBROOTCGAL/include '
configopts += '-DGMP_INCLUDE_DIR=$EBROOTGMP/include -DGMP_LIBRARIES=$EBROOTGMP/lib -DGMPXX_LIBRARIES=$EBROOTGMP/lib '
```

File: easybuild/easyconfigs/y/Yade/Yade-1.20.0-foss-2016a-Python-2.7.11.eb [hunk 15]
```diff
@@ -0,0 +1,55 @@
...
+    ('Boost', '1.60.0', versionsuffix),
+    ('GL2PS', '1.3.9', '-Mesa-11.2.1'),
+    ('Python', '2.7.11'),
+    ('IPython', '3.2.3', versionsuffix),
+    ('PyGTS', '0.3.1', versionsuffix),
+    ('Eigen', '3.2.7'),
+    ('minieigen', '0.5.3', versionsuffix),
+    ('Qt', '4.8.7', '-GLib-2.48.0'),
+    ('GTS', '0.7.6'),
+    ('libQGLViewer', '2.6.3', '-Mesa-11.2.1'),
```
Comment (boegel) [inline easybuild/easyconfigs/y/Yade/Yade-1.20.0-foss-2016a-Python-2.7.11.eb]: since you depend on Mesa in two places, I suggest you do this (to be able to change the Mesa version in a single place later):

```python
mesaver = '11.2.1'
dependencies = [
    ...
    ('GL2PS', '1.3.9', '-Mesa-%s' % mesaver),
    ...
    ('libQGLViewer', '2.6.3', '-Mesa-%s' % mesaver),
    ...
]
```

File: easybuild/easyconfigs/m/minieigen/minieigen-0.5.3_Eigen.patch [hunk 16]
```diff
@@ -0,0 +1,24 @@
+diff -ru minieigen-0.5.3.orig/setup.py minieigen-0.5.3/setup.py
```
Comment (boegel) [inline easybuild/easyconfigs/m/minieigen/minieigen-0.5.3_Eigen.patch]: @jrbosch please add a line of comment above this line explaining the patch, and mention yourself as an author:

```
correctly specify location of Eigen include files (provide via EasyBuild)
author: ...
```

File: easybuild/easyconfigs/m/minieigen/minieigen-0.5.3-foss-2016a-Python-2.7.11.eb [hunk 17]
```diff
@@ -0,0 +1,29 @@
...
+
+homepage = 'https://launchpad.net/minieigen/'
+description = """A small wrapper for core parts of EIgen, c++ library for linear algebra."""
+
+toolchain = {'name': 'foss', 'version': '2016a'}
+
+sources = [SOURCE_TAR_GZ]
+source_urls = [PYPI_SOURCE]
+
+patches = ['minieigen-0.5.3_Eigen.patch']
```
Comment (boegel) [inline easybuild/easyconfigs/m/minieigen/minieigen-0.5.3-foss-2016a-Python-2.7.11.eb]: avoid hardwiring the version here, use `%(version)s`

```python
patches = ['minieigen-%(version)s_Eigen.patch']
```

File: easybuild/easyconfigs/l/Loki/Loki-0.1.7-foss-2016a.eb [hunk 18]
```diff
@@ -0,0 +1,23 @@
...
+
+name = 'Loki'
+version = '0.1.7'
+
+homepage = 'http://loki-lib.sourceforge.net/'
+description = """ Loki is a C++ library of designs, containing flexible implementations of common design patterns and idioms. """
+
+toolchain = {'name': 'foss', 'version': '2016a'}
+
+installopts = "prefix=%(installdir)s"
```
Comment (boegel) [inline easybuild/easyconfigs/l/Loki/Loki-0.1.7-foss-2016a.eb]: move this down, below `skipsteps`

**Comment by boegel** [comment]
@jrbosch looks like this is ready to remove the `(WIP)` tag, and make it `(REVIEW)`?

**Comment by boegel** [comment]
@jrbosch the sanity check for Yade fails for me because the libraries are located in `<Yade_prefix>/lib64/yade-1.20.0/` rather than `<Yade_prefix>/lib/x86_64-linux-gnu/`, you should take this into account in the `sanity_check_paths`

File: easybuild/easyconfigs/y/Yade/Yade-1.20.0-foss-2016a-Python-2.7.11.eb [hunk 19]
```diff
@@ -0,0 +1,49 @@
...
+
+separate_build_dir = True
+
+configopts = '-DENABLE_GUI=OFF -DENABLE_VTK=OFF -DENABLE_LINSOLV=OFF -DENABLE_PFVFLOW=OFF '
+configopts += '-DCGAL_INCLUDE_DIR=$EBROOTCGAL/include '
+configopts += '-DGMP_INCLUDE_DIR=$EBROOTGMP/include -DGMP_LIBRARIES=$EBROOTGMP/lib -DGMPXX_LIBRARIES=$EBROOTGMP/lib '
+
+sanity_check_paths = {
+    'files': ['bin/yade-%(version)s', 'bin/yade-%(version)s-batch', 
+              'lib/x86_64-linux-gnu/yade-%%(version)s/libyade.%s' % SHLIB_EXT],
```
Comment (boegel) [inline easybuild/easyconfigs/y/Yade/Yade-1.20.0-foss-2016a-Python-2.7.11.eb]: seems like you'll need to change this to:

```
('lib/x86_64-linux-gnu/yade-%%(version)s/libyade.%s' % SHLIB_EXT, 'lib64/yade-%%(version)s/libyade.%s' % SHLIB_EXT),
```

if either of those is there, EB will be happy

File: easybuild/easyconfigs/y/Yade/Yade-1.20.0-foss-2016a-Python-2.7.11.eb [hunk 20]
```diff
@@ -0,0 +1,49 @@
...
+    'dirs': ['lib/x86_64-linux-gnu/yade-%(version)s/py/yade'],
```
Comment (boegel) [inline easybuild/easyconfigs/y/Yade/Yade-1.20.0-foss-2016a-Python-2.7.11.eb]: should be:

```python
dirs': [('lib/x86_64-linux-gnu/yade-%(version)s/py/yade', 'lib64/yade-%(version)s/py/yade')],
```

**Comment by boegel** [comment]
@jrbosch should the `lib` or `lib64` subdirectory also be added to `$LD_LIBRARY_PATH`?

Same for the `/py/` directory, should it be added to `$PYTHONPATH`?

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 2 out of 3 (3 easyconfigs in this PR)
node2608.swalot.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/c8e5ff0cfeeeac329d231edd6054c129 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 2 out of 3 (3 easyconfigs in this PR)
node2159.delcatty.os - Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/2744838578b8fa908550e446888c3849 for a full test report.

**Comment by jrbosch** [comment]
@boegel I just need bump the version of freetype in matplotlib easyconfig but I don't know if that is the correct way because matplotlib was merged so I created a new matplotlib easyconfig with freetype 2.6.3 dependency. The name matplotlib-1.5.1-foss-2016a-Python-2.7.11-freetype-2.6.3.eb is fine?

if is it better bump freetype version to 2.6.3 in matpoltlib easyconfig merged? I would do it.

everything works well !!!

**Comment by boegel** [comment]
@jrbosch bumping the freetype version in the existing easyconfig may lead to conflicts in other places, so I suggest you go with `matplotlib-1.5.1-foss-2016a-Python-2.7.11-freetype-2.6.3.eb`

File: easybuild/easyconfigs/m/matplotlib/matplotlib-1.5.1-foss-2016a-Python-2.7.11-freetype-2.6.3.eb [hunk 21]
```diff
@@ -0,0 +1,44 @@
+easyblock = 'Bundle'
+
+name = 'matplotlib'
+version = '1.5.1'
+versionsuffix = '-Python-%(pyver)s-freetype-2.6.3'
```
Comment (boegel) [inline easybuild/easyconfigs/m/matplotlib/matplotlib-1.5.1-foss-2016a-Python-2.7.11-freetype-2.6.3.eb]: try to avoid hardcoding the freetype version in two places:

```python
freetype_ver = '2.6.3'
versionsuffix = '-Python-%(pyver)s-freetype-%s' % freetype_ver
...
dependencies = [
    ...
    ('freetype', freetype_ver),
    ...
]
```

File: easybuild/easyconfigs/s/SuiteSparse/SuiteSparse-4.5.2-foss-2016a-METIS-5.1.0.eb [hunk 22]
```diff
@@ -0,0 +1,27 @@
...
+source_urls = ['http://faculty.cse.tamu.edu/davis/SuiteSparse/']
+sources = [SOURCE_TAR_GZ]
+
+metis = 'METIS'
+metis_ver = '5.1.0'
+versionsuffix = '-%s-%s' % (metis, metis_ver)
+dependencies = [(metis, metis_ver)]
+
+preconfigopts = 'MY_METIS_LIB="$EBROOTMETIS/lib/libmetis.so" ' 
+preconfigopts += 'LAPACK="$LIBLAPACK" '
```
Comment (boegel) [inline easybuild/easyconfigs/s/SuiteSparse/SuiteSparse-4.5.2-foss-2016a-METIS-5.1.0.eb]: why is this needed, the existing SuiteSparse easyconfigs don't have this?

File: easybuild/easyconfigs/m/matplotlib/matplotlib-1.5.1-foss-2016a-Python-2.7.11-freetype-2.6.3.eb [hunk 23]
```diff
@@ -0,0 +1,46 @@
...
+]
+
+exts_list = [
+    ('Cycler', '0.9.0', {
+        'modulename': 'cycler',
+        'source_urls': ['https://pypi.python.org/packages/source/C/Cycler'],
+        'source_tmpl': 'cycler-%(version)s.tar.gz',
+    }),
+    (name, version, {
+        'source_urls': ['https://pypi.python.org/packages/source/m/matplotlib'],
```
Comment (boegel) [inline easybuild/easyconfigs/m/matplotlib/matplotlib-1.5.1-foss-2016a-Python-2.7.11-freetype-2.6.3.eb]: please include this line below, which fixes a problem (segfault) with linking `matplotlib` to the EB-provided Tcl/Tk:

```python
'patches': ['matplotlib-1.5.1_fix-Tcl-Tk-libdir.patch']
```

cfr. #2971

File: easybuild/easyconfigs/s/SuiteSparse/SuiteSparse-4.5.2-foss-2016a-METIS-5.1.0.eb (hunk 22 above)
Comment (jrbosch) [inline easybuild/easyconfigs/s/SuiteSparse/SuiteSparse-4.5.2-foss-2016a-METIS-5.1.0.eb]: SuiteSparse easyconfig was taked from PR #2766 and I just add MY_METIS_LIB. If I don't include 'LAPACK="$LIBLAPACK" one error occur in building step because lapack.
But now I see that a rumtime error occur in Yade (problem with cholmod ) because I include SuiteSparse so I'm going to remove this dependency and disable the fluid coupling module in Yade that is unnecessary for now

Comment (boegel) [inline easybuild/easyconfigs/s/SuiteSparse/SuiteSparse-4.5.2-foss-2016a-METIS-5.1.0.eb]: ok, please remove this SuiteSparse easyconfig too then, since it'll need some work (cfr. https://github.com/hpcugent/easybuild-easyconfigs/pull/3050)

**Comment by jrbosch** [comment]
@boegel when you say:
"the lib or lib64 subdirectory also be added to $LD_LIBRARY_PATH?
Same for the /py/ directory, should it be added to $PYTHONPATH" 

- this is automatic done by the generic easyblock or should I do something with this?

**Comment by boegel** [comment]
@jrbosch the default mechanism only considers `lib` and `lib64`, not subdirectories

you can specify additional paths with:

```python
modextrapaths = {
    'LD_LIBRARY_PATH': ['lib64/lib64/yade-%(version)s', 'lib/x86_64-linux-gnu/yade-%(version)s'],
    'LIBRARY_PATH': ['lib64/lib64/yade-%(version)s', 'lib/x86_64-linux-gnu/yade-%(version)s'],
    'PYTHONPATH: ['lib/x86_64-linux-gnu/yade-%(version)s/py', 'lib64/yade-%(version)s/py'],
}

**Comment by jrbosch** [comment]
@boegel I made various runtime test and everything is fine.

File: easybuild/easyconfigs/m/matplotlib/matplotlib-1.5.1-foss-2016a-Python-2.7.11-freetype-2.6.3.eb (hunk 23 above)
Comment (boegel) [inline easybuild/easyconfigs/m/matplotlib/matplotlib-1.5.1-foss-2016a-Python-2.7.11-freetype-2.6.3.eb]: done, thanks @jr

File: easybuild/easyconfigs/v/VTK/VTK-6.3.0-foss-2016a-Python-2.7.11.eb [hunk 24]
```diff
@@ -0,0 +1,48 @@
...
+    ('libGLU', '9.0.0', '-Mesa-11.2.1'),
+    ('libXt', '1.1.5'),
+]
+
+configopts = "-DOPENGL_glu_LIBRARY=$EBROOTLIBGLU/lib/libGLU.%s " % SHLIB_EXT
+configopts += "-DOPENGL_gl_LIBRARY=$EBROOTMESA/lib/libGL.%s " % SHLIB_EXT
+configopts += "-DOPENGL_INCLUDE_DIR=$EBROOTMESA/include -DVTK_WRAP_PYTHON=ON "
+configopts += "-DPYTHON_LIBRARY=$EBROOTPYTHON/lib/libpython%%(pyshortver)s.%s " % SHLIB_EXT
+configopts += " -DPYTHON_INCLUDE_DIR=$EBROOTPYTHON/include/python%(pyshortver)s "
+preinstallopts = "mkdir -p %(installdir)s/lib/python2.7/site-packages/ && "
```
Comment (boegel) [inline easybuild/easyconfigs/v/VTK/VTK-6.3.0-foss-2016a-Python-2.7.11.eb]: please use `%(pyshortver)s` rather than hardcoding `2.7` (I'll fix this for the other existing VTK easyconfigs in a separate PR):

```python
preinstallopts = "mkdir -p %(installdir)s/lib/python%(pyshortver)s/site-packages/ && "
```

File: easybuild/easyconfigs/m/minieigen/minieigen-0.5.3-foss-2016a-Python-2.7.11.eb [hunk 25]
```diff
@@ -0,0 +1,26 @@
...
+source_urls = [PYPI_SOURCE]
+
+dependencies = [
+    ('Boost', '1.60.0', versionsuffix),
+    ('Eigen', '3.2.7'),
+    ('Python', '2.7.11'),
+]
+
+sanity_check_paths = {
+    'files': ['lib/python2.7/site-packages/minieigen.%s' % SHLIB_EXT],
```
Comment (boegel) [inline easybuild/easyconfigs/m/minieigen/minieigen-0.5.3-foss-2016a-Python-2.7.11.eb]: please use `%(pyshortver)s` rather than hardcoding `2.7`

```python
    'files': ['lib/python%%(pyshortver)s/site-packages/minieigen.%s' % SHLIB_EXT],
```

File: easybuild/easyconfigs/v/VTK/VTK-6.3.0-foss-2016a-Python-2.7.11.eb (hunk 24 above)
Comment (boegel) [inline easybuild/easyconfigs/v/VTK/VTK-6.3.0-foss-2016a-Python-2.7.11.eb]: see #3052

**Comment by jrbosch** [comment]
@boegel done!

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in this PR)
node2433.golett.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/1b4626921383e13bfc132ea53b96b183 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in this PR)
node2159.delcatty.os - Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/c73cc66d7f34b1243d1f14be7797356a for a full test report.

**Comment by boegel** [comment]
Great, thanks @jrbosch, lgtm now!

I gave this whole PR another full test (cfr. latest test reports), and the builds are working fine on my end too now, so it seems like we've finally reached the point we were aiming for... Support for building and installing Yade and all dependencies will be included in EasyBuild v2.8.0 (planned for release next week).

Thanks *a lot* for your efforts on this @jrbosch, and for your patience in dealing with remarks and feedback to your contributions!
